p4-16/psa/psa.p4

```diff
@@ -568,12 +579,13 @@ control ComputeChecksum<H, M>(inout H hdr, inout M user_meta);
 
 control Deparser<H>(packet_out buffer, in H hdr);
 
-package PSA_Switch<H, M>(Parser<H, M> p,
-                         VerifyChecksum<H, M> vr,
+package PSA_Switch<H, M>(IngressParser<H, M> ip,
                          Ingress<H, M> ig,
+                         Deparser<H> igdep,
+                         EgressParser<H, M> ep,
                          Egress<H, M> eg,
                          ComputeChecksum<H, M> ck,
-                         Deparser<H> dep);
+                         Deparser<H> egdep);
```

Comment:
If a programmer decides to put checksum calculation before the last place that they modify the relevant headers, then the programmer has created a problem, and must live with their mistake.  If we allow checksum calculation in a deparser, we are saying "do checksum fixups before the emit calls, or it will be incorrect."  That seems roughly the same as allowing it in ingress/egress, and saying "do checksum fixups after all header modifications, or it will be incorrect."

I've added computeChecksum back in, but this time both for ingress and egress.  Eventually we will stop changing this.